src/GenAPI/Microsoft.DotNet.GenAPI/CSharpFileBuilder.cs

```diff
@@ -128,6 +130,104 @@ private void Visit(IAssemblySymbol assembly)
             return namespaceNode;
         }
 
+        private static string GetNonInterfaceName(string name)
+        {
+            int idx = name.LastIndexOf('.');
+            return (idx > 0) ? name.Substring(idx + 1) : name;
+        }
+
+        // Compare the equality of two method signatures for the purpose of emitting a "new"
+        // keyword on a method's return type. This is *not* meant to be complete implementation,
+        // but rather a heuristic to check that one method may hide another.
+        private static bool SignatureEquals(IMethodSymbol? method, IMethodSymbol? baseMethod)
+        {
+            if (method == null || baseMethod == null)
+            {
+                return false;
+            }
+
+            if (method.Equals(baseMethod, SymbolEqualityComparer.Default))
+            {
+                return true;
+            }
+
+            if (GetNonInterfaceName(method.Name) != GetNonInterfaceName(baseMethod.Name))
+            {
+                return false;
+            }
+
+            if (method.Arity != baseMethod.Arity || method.Parameters.Length != baseMethod.Parameters.Length)
+            {
+                return false;
+            }
+
+            // compare parameter types
+            for (int i = 0; i < method.Parameters.Length; i++)
+            {
+                if (!method.Parameters[i].Type.Equals(baseMethod.Parameters[i].Type, SymbolEqualityComparer.Default))
+                {
+                    return false;
+                }
+            }
+
+            if (method.IsVararg != baseMethod.IsVararg)
```

Comment:
Interesting.  Here is how this might surface in this scenario.
https://sharplab.io/#v2:EYLgHgbALANALiAlgGxgExAagD4AEBMAjALABQZuAzAAQHUDC1A3mdW7TblNQGICuAOwAUAfREBDAE4BzZIgDOcAJSt2LUu2oBfMjtJVa+agBFqIBs1VsDAgKYB3Wt37CxU2QuXNtVjk96CQgAOUuIAtvLUAPbAAFa2AMZwANoAutQAbuLIfLbySt56mgZcAcKIAnDUAGZRUQVMPhrsJc6BDU1aQA===

Reply:
It seems that GenAPI currently does not output `__arglist` from the metadata.
Adding a test case for this scenario would fail, since `__arglist` wouldn't even be printed. I will document this, and open up an issue.